projects/max14906/src/platform/maxim/main.c

```diff
@@ -58,11 +58,30 @@ int main()
 
 #ifdef BASIC_EXAMPLE
 	struct no_os_uart_desc *uart_desc;
+	struct no_os_irq_ctrl_desc *gpio_irq_desc;
```

Comment:
I think all the gpio irq related code that you put in this maxim/main.c file should go to basic_example.c


And in parameters.h/.c (bsp) you should provide GPIO_IRQ_ID, GPIO_IRQ_OPS and some common macro for GPIO0_IRQn and it's equivalent for stm32.

Reply:
There are discrepancies between maxim and stm when treating interrupts and I think these changes currently are related only to maxim platform

Reply:
ok, in that case we keep them as they are but we need to fix the discrepancies separately and i would say ASAP because we keep adding projects that we will need to refactor...